wallet/handler.go

```diff
@@ -259,6 +266,75 @@ func DescribeKey(store Store, req *DescribeKeyRequest) (*DescribeKeyResponse, er
 	return resp, nil
 }
 
+type RotateKeyRequest struct {
+	Wallet            string
+	Passphrase        string
+	NewPublicKey      string
+	TXBlockHeight     uint32
+	TargetBlockHeight uint32
+}
+
+type RotateKeyResponse struct {
+	MasterPublicKey   string `json:"master_public_key"`
+	NewPublicKey      string `json:"new_public_key"`
+	Base64Transaction string `json:"base64_transaction"`
+}
+
+func RotateKey(store Store, req *RotateKeyRequest) (*RotateKeyResponse, error) {
+	w, err := getWallet(store, req.Wallet, req.Passphrase)
+	if err != nil {
+		return nil, err
+	}
+
+	mKeyPair, err := w.GetMasterKeyPair()
+	if err != nil {
+		return nil, err
+	}
+
+	newKeyPair, err := w.DescribeKeyPair(req.NewPublicKey)
```

Comment:
We only need the public-key here and not the key-pair. Can you add `Index()` to `type PublicKey interface` and use `w.DescribePublicKey()` instead.